### PR TITLE
[browser][wasm] JavaScript core Map object disposal - Do Not Merge

### PR DESCRIPTION
**Pull request by kjpou1**
This fixes a crash that was happening when garbage collection kicked in.  When allocating larger key value pair counts the Map object would throw a NRE intermittently.  

Also, added larger iteration tests to the map tests so this can be caught in the future.

There seems to be another problem somewhere and needs to be looked at further.

### CODE REVIEW

**Comment by campersau** [comment]
Was that the cause of https://github.com/dotnet/runtime/pull/39061 so `MapTests.cs` can be enabled again?
https://github.com/dotnet/runtime/blob/9745dd7e7dba725faed70a0258f5965557398a0f/src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.Tests.csproj#L15

**Comment by kjpou1** [comment]
@marek-safar could you take a look ?

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Map.cs [hunk 1]
```diff
@@ -147,16 +147,9 @@ private void Dispose(bool disposing)
                 }
             }
 
-            ~MapEnumerator()
```
Comment (marek-safar) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Map.cs:150]: I don't understand how can this fix the problem. It only disables the tests coverage,

Comment (kjpou1) [reply src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Map.cs:150]: The tests were failing with null object when GC kicked in.  The taking this out it the null object error went away when GC kicked in.

**Review by lewing — COMMENTED** [review]
The dispose logic doesn't make much sense now, this is a sealed class with only a public Dispose method and an indirect reference to an an unmanaged object that is guarded against reset in a strange way.  Can you explain why removing the finalizer is the correct fix?

**Comment by kjpou1** [comment]
Worked on the disposing logic but there seems to be another error. 

```
JS exception: Error: no idea on how to unbox value types
        
  dbug: test[0]
        Error: no idea on how to unbox value types
        
  dbug: test[0]
            at Object.unbox_mono_obj (dotnet.js:7840:13)
```

**Comment by lewing** [comment]
@kjpou1 if this is active please rebase it otherwise please close the pr

**Comment by kjpou1** [comment]
No longer valid as other PR resolved issues and map tests are enabled again.

**Pull request closed without merging**